# DATA CLEANING

In [1]:
import pandas as pd


In [2]:
data=pd.read_csv(r"D:\sales_data.csv")

In [3]:
data

order_id  order_date   ship_date       ship_mode  \
0         AG-2011-2040  01-01-2011  01-06-2011  Standard Class   
1        IN-2011-47883  01-01-2011  01-08-2011  Standard Class   
2         HU-2011-1220  01-01-2011  01-05-2011    Second Class   
3      IT-2011-3647632  01-01-2011  01-05-2011    Second Class   
4        IN-2011-47883  01-01-2011  01-08-2011  Standard Class   
...                ...         ...         ...             ...   
51285   CA-2014-115427  12/31/2014  01-04-2015  Standard Class   
51286     MO-2014-2560  12/31/2014  01-05-2015  Standard Class   
51287   MX-2014-110527  12/31/2014  01-02-2015    Second Class   
51288   MX-2014-114783  12/31/2014  01-06-2015  Standard Class   
51289   CA-2014-156720  12/31/2014  01-04-2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ...         category sub_category  \
0       Africa  ...  Office Supplies      Storage   
1      Oceania  ...  Office Supplies     Supplies   
2         EMEA  ...  Office Supplies      Storage   
3        North  ...  Office Supplies        Paper   
4      Oceania  ...        Furniture  Furnishings   
...        ...  ...              ...          ...   
51285     West  ...  Office Supplies      Binders   
51286   Africa  ...  Office Supplies      Binders   
51287  Central  ...  Office Supplies       Labels   
51288    North  ...  Office Supplies       Labels   
51289     West  ...  Office Supplies    Fasteners   

                                          product_name sales quantity  \
0                                  Tenex Lockers, Blue   408        2   
1                             Acme Trimmer, High Speed   120        3   
2                              Tenex Box, Single Width    66        4   
3                          Enermax Note Cards, Premium    45        3   
4                           Eldon Light Bulb, Duo Pack   114        5   
...                                                ...   ...      ...   
51285  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl    14        2   
51286          Wilson Jones Hole Reinforcements, Clear     4        1   
51287           Hon Color Coded Labels, 5000 Label Set    26        3   
51288           Hon Legal Exhibit Labels, Alphabetical     7        1   
51289                              Bagged Rubber Bands     3        3   

       discount    profit  shipping_cost  order_priority  year;  
0           0.0  106.1400          35.46          Medium  2011;  
1           0.1   36.0360           9.72          Medium  2011;  
2           0.0   29.6400           8.17            High  2011;  
3           0.5  -26.0550           4.82            High  2011;  
4           0.1   37.7700           4.70          Medium  2011;  
...         ...       ...            ...             ...    ...  
51285       0.2    4.5188           0.89          Medium  2014;  
51286       0.0    0.4200           0.49          Medium  2014;  
51287       0.0   12.3600           0.35          Medium  2014;  
51288       0.0    0.5600           0.20          Medium  2014;  
51289   

## checking info

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51290 entries, 0 to 51289
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   order_id        51290 non-null  object 
 1   order_date      51290 non-null  object 
 2   ship_date       51290 non-null  object 
 3   ship_mode       51290 non-null  object 
 4   customer_name   51290 non-null  object 
 5   segment         51290 non-null  object 
 6   state           51290 non-null  object 
 7   country         51290 non-null  object 
 8   market          51290 non-null  object 
 9   region          51290 non-null  object 
 10  product_id      51290 non-null  object 
 11  category        51290 non-null  object 
 12  sub_category    51290 non-null  object 
 13  product_name    51290 non-null  object 
 14  sales           51290 non-null  object 
 15  quantity        51290 non-null  int64  
 16  discount        51290 non-null  float64
 17  profit          51290 non-null 

## finding null values

In [5]:
data.isnull().sum()

order_id          0
order_date        0
ship_date         0
ship_mode         0
customer_name     0
segment           0
state             0
country           0
market            0
region            0
product_id        0
category          0
sub_category      0
product_name      0
sales             0
quantity          0
discount          0
profit            0
shipping_cost     0
order_priority    0
year;             0
dtype: int64

In [6]:
data.tail()

order_id  order_date   ship_date       ship_mode  \
51285  CA-2014-115427  12/31/2014  01-04-2015  Standard Class   
51286    MO-2014-2560  12/31/2014  01-05-2015  Standard Class   
51287  MX-2014-110527  12/31/2014  01-02-2015    Second Class   
51288  MX-2014-114783  12/31/2014  01-06-2015  Standard Class   
51289  CA-2014-156720  12/31/2014  01-04-2015  Standard Class   

          customer_name    segment             state        country  market  \
51285        Erica Bern  Corporate        California  United States      US   
51286         Liz Preis   Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton   Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen   Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias   Consumer          Colorado  United States      US   

        region  ...         category sub_category  \
51285     West  ...  Office Supplies      Binders   
51286   Africa  ...  Office Supplies      Binders   
51287  Central  ...  Office Supplies       Labels   
51288    North  ...  Office Supplies       Labels   
51289     West  ...  Office Supplies    Fasteners   

                                          product_name sales quantity  \
51285  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl    14        2   
51286          Wilson Jones Hole Reinforcements, Clear     4        1   
51287           Hon Color Coded Labels, 5000 Label Set    26        3   
51288           Hon Legal Exhibit Labels, Alphabetical     7        1   
51289                              Bagged Rubber Bands     3        3   

       discount   profit  shipping_cost  order_priority  year;  
51285       0.2   4.5188           0.89          Medium  2014;  
51286       0.0   0.4200           0.49          Medium  2014;  
51287       0.0  12.3600           0.35          Medium  2014;  
51288       0.0   0.5600           0.20          Medium  2014;  
51289       0.2  -0.6048           0.17          Medium  2014;  

[5 rows x 21 columns]

In [7]:
#checking the shape of data
data.shape

(51290, 21)

### for column order_id

In [8]:
data["order_id"]

0           AG-2011-2040
1          IN-2011-47883
2           HU-2011-1220
3        IT-2011-3647632
4          IN-2011-47883
              ...       
51285     CA-2014-115427
51286       MO-2014-2560
51287     MX-2014-110527
51288     MX-2014-114783
51289     CA-2014-156720
Name: order_id, Length: 51290, dtype: object

In [9]:
# checking the unique values 
data["order_id"].unique()

array(['AG-2011-2040', 'IN-2011-47883', 'HU-2011-1220', ...,
       'MX-2014-108574', 'MO-2014-2560', 'CA-2014-156720'], dtype=object)

In [10]:
data["order_id"].nunique()

25035

### for column order_date


In [11]:
data["order_date"].dtypes

dtype('O')

In [12]:
data["order_date"].unique()

array(['01-01-2011', '01-02-2011', '01-03-2011', ..., '12/29/2014',
       '12/30/2014', '12/31/2014'], dtype=object)

* The order of date is not in standard format.

In [13]:
data[data["order_date"].str.contains("-")]

order_id  order_date   ship_date       ship_mode  \
0         AG-2011-2040  01-01-2011  01-06-2011  Standard Class   
1        IN-2011-47883  01-01-2011  01-08-2011  Standard Class   
2         HU-2011-1220  01-01-2011  01-05-2011    Second Class   
3      IT-2011-3647632  01-01-2011  01-05-2011    Second Class   
4        IN-2011-47883  01-01-2011  01-08-2011  Standard Class   
...                ...         ...         ...             ...   
50060   US-2014-125213  12-12-2014  12/14/2014     First Class   
50061     TU-2014-7890  12-12-2014  12/19/2014  Standard Class   
50062   CA-2014-165099  12-12-2014  12/14/2014     First Class   
50063     NI-2014-9730  12-12-2014  12/16/2014  Standard Class   
50064     NI-2014-9730  12-12-2014  12/16/2014  Standard Class   

           customer_name      segment            state        country  market  \
0        Toby Braunhardt     Consumer      Constantine        Algeria  Africa   
1            Joseph Holt     Consumer  New South Wales      Australia    APAC   
2          Annie Thurman     Consumer         Budapest        Hungary    EMEA   
3           Eugene Moren  Home Office        Stockholm         Sweden      EU   
4            Joseph Holt     Consumer  New South Wales      Australia    APAC   
...                  ...          ...              ...            ...     ...   
50060          Nona Balk    Corporate       New Jersey  United States      US   
50061   Jennifer Jackson     Consumer         Istanbul         Turkey    EMEA   
50062        Dennis Kane     Consumer            Texas  United States      US   
50063  Alan Schoenberger    Corporate            Lagos        Nigeria  Africa   
50064  Alan Schoenberger    Corporate            Lagos        Nigeria  Africa   

        region  ...         category sub_category  \
0       Africa  ...  Office Supplies      Storage   
1      Oceania  ...  Office Supplies     Supplies   
2         EMEA  ...  Office Supplies      Storage   
3        North  ...  Office Supplies        Paper   
4      Oceania  ...        Furniture  Furnishings   
...        ...  ...              ...          ...   
50060     East  ...  Office Supplies    Fasteners   
50061     EMEA  ...  Office Supplies       Labels   
50062  Central  ...  Office Supplies   Appliances   
50063   Africa  ...  Office Supplies      Binders   
50064   Africa  ...  Office Supplies      Binders   

                                       product_name sales quantity  discount  \
0                               Tenex Lockers, Blue   408        2       0.0   
1                          Acme Trimmer, High Speed   120        3       0.1   
2                           Tenex Box, Single Width    66        4       0.0   
3                       Enermax Note Cards, Premium    45        3       0.5   
4                        Eldon Light Bulb, Duo Pack   114        5       0.1   
...                                             ...   ...      ...       ...   
50060                           Stockwell Push Pins     7        3       0.0   
50061          Avery File Folder Labels, Adjustable     3        1       0.6   
50062  Hoover Commercial Lightweight Upright Vacuum     1        2       0.8   
50063             Acco Hole Reinforcements, Durable     2        1       0.7   
50064                        Ibico Index Tab, Clear     3        1       0.7   

         profit  shipping_cost  order_priority  year;  
0      106.1400          35.46          Medium  2011;  
1       36.0360           9.72          Medium  2011;  
2       29.6400           8.17            High  2011;  
3      -26.0550           4.82            High  2011;  
4       37.7700           4.70          Medium  2011;  
...         ...            ...             ...    ...  
50060    2.1582           0.34            High  2014;  
50061   -1.6920           0.33             Low  2014;  
50062   -3.7584           0.14            High  2014;  
50063   -1.9020           0.06          Medium  2014;  
50064   -4.6470           0.05    

* There are total 20067 rows which contains wrong date format.

In [14]:
# replacing the "-" with "/"

data["order_date"]=data["order_date"].str.replace("-","/")

In [15]:
data

order_id  order_date   ship_date       ship_mode  \
0         AG-2011-2040  01/01/2011  01-06-2011  Standard Class   
1        IN-2011-47883  01/01/2011  01-08-2011  Standard Class   
2         HU-2011-1220  01/01/2011  01-05-2011    Second Class   
3      IT-2011-3647632  01/01/2011  01-05-2011    Second Class   
4        IN-2011-47883  01/01/2011  01-08-2011  Standard Class   
...                ...         ...         ...             ...   
51285   CA-2014-115427  12/31/2014  01-04-2015  Standard Class   
51286     MO-2014-2560  12/31/2014  01-05-2015  Standard Class   
51287   MX-2014-110527  12/31/2014  01-02-2015    Second Class   
51288   MX-2014-114783  12/31/2014  01-06-2015  Standard Class   
51289   CA-2014-156720  12/31/2014  01-04-2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ...         category sub_category  \
0       Africa  ...  Office Supplies      Storage   
1      Oceania  ...  Office Supplies     Supplies   
2         EMEA  ...  Office Supplies      Storage   
3        North  ...  Office Supplies        Paper   
4      Oceania  ...        Furniture  Furnishings   
...        ...  ...              ...          ...   
51285     West  ...  Office Supplies      Binders   
51286   Africa  ...  Office Supplies      Binders   
51287  Central  ...  Office Supplies       Labels   
51288    North  ...  Office Supplies       Labels   
51289     West  ...  Office Supplies    Fasteners   

                                          product_name sales quantity  \
0                                  Tenex Lockers, Blue   408        2   
1                             Acme Trimmer, High Speed   120        3   
2                              Tenex Box, Single Width    66        4   
3                          Enermax Note Cards, Premium    45        3   
4                           Eldon Light Bulb, Duo Pack   114        5   
...                                                ...   ...      ...   
51285  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl    14        2   
51286          Wilson Jones Hole Reinforcements, Clear     4        1   
51287           Hon Color Coded Labels, 5000 Label Set    26        3   
51288           Hon Legal Exhibit Labels, Alphabetical     7        1   
51289                              Bagged Rubber Bands     3        3   

       discount    profit  shipping_cost  order_priority  year;  
0           0.0  106.1400          35.46          Medium  2011;  
1           0.1   36.0360           9.72          Medium  2011;  
2           0.0   29.6400           8.17            High  2011;  
3           0.5  -26.0550           4.82            High  2011;  
4           0.1   37.7700           4.70          Medium  2011;  
...         ...       ...            ...             ...    ...  
51285       0.2    4.5188           0.89          Medium  2014;  
51286       0.0    0.4200           0.49          Medium  2014;  
51287       0.0   12.3600           0.35          Medium  2014;  
51288       0.0    0.5600           0.20          Medium  2014;  
51289   

In [16]:
data

order_id  order_date   ship_date       ship_mode  \
0         AG-2011-2040  01/01/2011  01-06-2011  Standard Class   
1        IN-2011-47883  01/01/2011  01-08-2011  Standard Class   
2         HU-2011-1220  01/01/2011  01-05-2011    Second Class   
3      IT-2011-3647632  01/01/2011  01-05-2011    Second Class   
4        IN-2011-47883  01/01/2011  01-08-2011  Standard Class   
...                ...         ...         ...             ...   
51285   CA-2014-115427  12/31/2014  01-04-2015  Standard Class   
51286     MO-2014-2560  12/31/2014  01-05-2015  Standard Class   
51287   MX-2014-110527  12/31/2014  01-02-2015    Second Class   
51288   MX-2014-114783  12/31/2014  01-06-2015  Standard Class   
51289   CA-2014-156720  12/31/2014  01-04-2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ...         category sub_category  \
0       Africa  ...  Office Supplies      Storage   
1      Oceania  ...  Office Supplies     Supplies   
2         EMEA  ...  Office Supplies      Storage   
3        North  ...  Office Supplies        Paper   
4      Oceania  ...        Furniture  Furnishings   
...        ...  ...              ...          ...   
51285     West  ...  Office Supplies      Binders   
51286   Africa  ...  Office Supplies      Binders   
51287  Central  ...  Office Supplies       Labels   
51288    North  ...  Office Supplies       Labels   
51289     West  ...  Office Supplies    Fasteners   

                                          product_name sales quantity  \
0                                  Tenex Lockers, Blue   408        2   
1                             Acme Trimmer, High Speed   120        3   
2                              Tenex Box, Single Width    66        4   
3                          Enermax Note Cards, Premium    45        3   
4                           Eldon Light Bulb, Duo Pack   114        5   
...                                                ...   ...      ...   
51285  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl    14        2   
51286          Wilson Jones Hole Reinforcements, Clear     4        1   
51287           Hon Color Coded Labels, 5000 Label Set    26        3   
51288           Hon Legal Exhibit Labels, Alphabetical     7        1   
51289                              Bagged Rubber Bands     3        3   

       discount    profit  shipping_cost  order_priority  year;  
0           0.0  106.1400          35.46          Medium  2011;  
1           0.1   36.0360           9.72          Medium  2011;  
2           0.0   29.6400           8.17            High  2011;  
3           0.5  -26.0550           4.82            High  2011;  
4           0.1   37.7700           4.70          Medium  2011;  
...         ...       ...            ...             ...    ...  
51285       0.2    4.5188           0.89          Medium  2014;  
51286       0.0    0.4200           0.49          Medium  2014;  
51287       0.0   12.3600           0.35          Medium  2014;  
51288       0.0    0.5600           0.20          Medium  2014;  
51289   

### for column ship_date

In [17]:
# checking the info of this column
data["ship_date"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 51290 entries, 0 to 51289
Series name: ship_date
Non-Null Count  Dtype 
--------------  ----- 
51290 non-null  object
dtypes: object(1)
memory usage: 400.8+ KB


In [18]:
data["ship_date"].unique()

array(['01-06-2011', '01-08-2011', '01-05-2011', ..., '01-05-2015',
       '01-06-2015', '01-07-2015'], dtype=object)

In [19]:
# replacing the string "-" with "/".
data["ship_date"]=data["ship_date"].str.replace("-","/")

In [20]:
data

order_id  order_date   ship_date       ship_mode  \
0         AG-2011-2040  01/01/2011  01/06/2011  Standard Class   
1        IN-2011-47883  01/01/2011  01/08/2011  Standard Class   
2         HU-2011-1220  01/01/2011  01/05/2011    Second Class   
3      IT-2011-3647632  01/01/2011  01/05/2011    Second Class   
4        IN-2011-47883  01/01/2011  01/08/2011  Standard Class   
...                ...         ...         ...             ...   
51285   CA-2014-115427  12/31/2014  01/04/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  01/05/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  01/02/2015    Second Class   
51288   MX-2014-114783  12/31/2014  01/06/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  01/04/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ...         category sub_category  \
0       Africa  ...  Office Supplies      Storage   
1      Oceania  ...  Office Supplies     Supplies   
2         EMEA  ...  Office Supplies      Storage   
3        North  ...  Office Supplies        Paper   
4      Oceania  ...        Furniture  Furnishings   
...        ...  ...              ...          ...   
51285     West  ...  Office Supplies      Binders   
51286   Africa  ...  Office Supplies      Binders   
51287  Central  ...  Office Supplies       Labels   
51288    North  ...  Office Supplies       Labels   
51289     West  ...  Office Supplies    Fasteners   

                                          product_name sales quantity  \
0                                  Tenex Lockers, Blue   408        2   
1                             Acme Trimmer, High Speed   120        3   
2                              Tenex Box, Single Width    66        4   
3                          Enermax Note Cards, Premium    45        3   
4                           Eldon Light Bulb, Duo Pack   114        5   
...                                                ...   ...      ...   
51285  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl    14        2   
51286          Wilson Jones Hole Reinforcements, Clear     4        1   
51287           Hon Color Coded Labels, 5000 Label Set    26        3   
51288           Hon Legal Exhibit Labels, Alphabetical     7        1   
51289                              Bagged Rubber Bands     3        3   

       discount    profit  shipping_cost  order_priority  year;  
0           0.0  106.1400          35.46          Medium  2011;  
1           0.1   36.0360           9.72          Medium  2011;  
2           0.0   29.6400           8.17            High  2011;  
3           0.5  -26.0550           4.82            High  2011;  
4           0.1   37.7700           4.70          Medium  2011;  
...         ...       ...            ...             ...    ...  
51285       0.2    4.5188           0.89          Medium  2014;  
51286       0.0    0.4200           0.49          Medium  2014;  
51287       0.0   12.3600           0.35          Medium  2014;  
51288       0.0    0.5600           0.20          Medium  2014;  
51289   

* Successfully replaced the "-" with "/".

### for column ship_mode

In [21]:
# checking the info
data["ship_mode"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 51290 entries, 0 to 51289
Series name: ship_mode
Non-Null Count  Dtype 
--------------  ----- 
51290 non-null  object
dtypes: object(1)
memory usage: 400.8+ KB


In [22]:
data["ship_mode"].unique()

array(['Standard Class', 'Second Class', 'Same Day', 'First Class'],
      dtype=object)

* There are 4 unique values present in it.


In [23]:
#  counting the value based on data
data["ship_mode"].value_counts()

ship_mode
Standard Class    30775
Second Class      10309
First Class        7505
Same Day           2701
Name: count, dtype: int64

* In the data, There are maximum numbers of counts for standard class i.e., 30775.
* There are 10309 numbers of counts for Second class.
* There are 7505 numbers of counts for First class.
* There are minimum numbers of counts for Same day i.e.,2701.

### for customer_name columns


In [24]:
# checking the info
data["customer_name"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 51290 entries, 0 to 51289
Series name: customer_name
Non-Null Count  Dtype 
--------------  ----- 
51290 non-null  object
dtypes: object(1)
memory usage: 400.8+ KB


In [25]:
data["customer_name"].values

array(['Toby Braunhardt', 'Joseph Holt', 'Annie Thurman', ...,
       'Charlotte Melton', 'Tamara Dahlen', 'Jill Matthias'], dtype=object)

In [26]:
data["customer_name"].unique()

array(['Toby Braunhardt', 'Joseph Holt', 'Annie Thurman', 'Eugene Moren',
       'Magdelene Morse', 'Kean Nguyen', 'Ken Lonsdale',
       'Lindsay Williams', 'Larry Blacks', 'Dorothy Dickinson',
       'Dennis Pardue', 'Stewart Visinsky', "Jas O'Carroll",
       'Chris McAfee', 'Nat Gilpin', 'Jack Garza', 'Russell Applegate',
       'Maya Herman', 'Beth Thompson', 'Arthur Prichep',
       'Arthur Wiediger', 'Grant Thornton', 'Darren Powers',
       'Bradley Drucker', 'Jasper Cacioppo', 'Jennifer Halladay',
       'Michael Granlund', 'Phillina Ober', 'Shaun Weien',
       'Shahid Hopkins', 'Brosina Hoffman', 'Robert Dilbeck',
       'Grace Kelly', 'Khloe Miller', 'Mick Brown', 'Maria Etezadi',
       'Maribeth Schnelling', 'Dave Poirier', 'Harry Marie',
       'Lycoris Saunders', 'Alyssa Crouse', "Jack O'Briant",
       'Adam Bellavance', 'Joy Smith', 'Henia Zydlo', 'Nicole Fjeld',
       'Christopher Martinez', 'Vivek Sundaresam', 'Joe Elijah',
       'Ed Braxton', 'Alejandro Grove', '

* There are few name in dataset with double cots.

In [27]:
#incomplete customer_name columns

### for column product_name


In [28]:
data["product_name"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 51290 entries, 0 to 51289
Series name: product_name
Non-Null Count  Dtype 
--------------  ----- 
51290 non-null  object
dtypes: object(1)
memory usage: 400.8+ KB


In [29]:
# checking the unique 
data["product_name"].unique()

array(['Tenex Lockers, Blue', 'Acme Trimmer, High Speed',
       'Tenex Box, Single Width', ...,
       'Panasonic BusinessÿTelephonesÿKX-T7736',
       'Bush Saratoga Collection 5-Shelf Bookcase, Hanover Cherry, *Special Order',
       'Acco Glide Clips'], dtype=object)

In [30]:
data["product_name"].nunique()

3788

In [31]:
#replacements = {'*': '', 'ÿ': ' '}
data["product_name"]=data["product_name"].str.replace("[*ÿ]","",regex=True)

In [32]:
data

order_id  order_date   ship_date       ship_mode  \
0         AG-2011-2040  01/01/2011  01/06/2011  Standard Class   
1        IN-2011-47883  01/01/2011  01/08/2011  Standard Class   
2         HU-2011-1220  01/01/2011  01/05/2011    Second Class   
3      IT-2011-3647632  01/01/2011  01/05/2011    Second Class   
4        IN-2011-47883  01/01/2011  01/08/2011  Standard Class   
...                ...         ...         ...             ...   
51285   CA-2014-115427  12/31/2014  01/04/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  01/05/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  01/02/2015    Second Class   
51288   MX-2014-114783  12/31/2014  01/06/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  01/04/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ...         category sub_category  \
0       Africa  ...  Office Supplies      Storage   
1      Oceania  ...  Office Supplies     Supplies   
2         EMEA  ...  Office Supplies      Storage   
3        North  ...  Office Supplies        Paper   
4      Oceania  ...        Furniture  Furnishings   
...        ...  ...              ...          ...   
51285     West  ...  Office Supplies      Binders   
51286   Africa  ...  Office Supplies      Binders   
51287  Central  ...  Office Supplies       Labels   
51288    North  ...  Office Supplies       Labels   
51289     West  ...  Office Supplies    Fasteners   

                                          product_name sales quantity  \
0                                  Tenex Lockers, Blue   408        2   
1                             Acme Trimmer, High Speed   120        3   
2                              Tenex Box, Single Width    66        4   
3                          Enermax Note Cards, Premium    45        3   
4                           Eldon Light Bulb, Duo Pack   114        5   
...                                                ...   ...      ...   
51285  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl    14        2   
51286          Wilson Jones Hole Reinforcements, Clear     4        1   
51287           Hon Color Coded Labels, 5000 Label Set    26        3   
51288           Hon Legal Exhibit Labels, Alphabetical     7        1   
51289                              Bagged Rubber Bands     3        3   

       discount    profit  shipping_cost  order_priority  year;  
0           0.0  106.1400          35.46          Medium  2011;  
1           0.1   36.0360           9.72          Medium  2011;  
2           0.0   29.6400           8.17            High  2011;  
3           0.5  -26.0550           4.82            High  2011;  
4           0.1   37.7700           4.70          Medium  2011;  
...         ...       ...            ...             ...    ...  
51285       0.2    4.5188           0.89          Medium  2014;  
51286       0.0    0.4200           0.49          Medium  2014;  
51287       0.0   12.3600           0.35          Medium  2014;  
51288       0.0    0.5600           0.20          Medium  2014;  
51289   

In [33]:
data["product_name"].unique()

array(['Tenex Lockers, Blue', 'Acme Trimmer, High Speed',
       'Tenex Box, Single Width', ...,
       'Panasonic BusinessTelephonesKX-T7736',
       'Bush Saratoga Collection 5-Shelf Bookcase, Hanover Cherry, Special Order',
       'Acco Glide Clips'], dtype=object)

* special characters replaced with "" whitespace.

### for column year

In [42]:
data["year"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 51290 entries, 0 to 51289
Series name: year
Non-Null Count  Dtype 
--------------  ----- 
51290 non-null  object
dtypes: object(1)
memory usage: 400.8+ KB


In [39]:

data.drop(columns="year;",inplace=True,axis=1)

In [35]:
data["year"]=data["year;"].str.replace(";","",regex=True)

In [41]:
data

order_id  order_date   ship_date       ship_mode  \
0         AG-2011-2040  01/01/2011  01/06/2011  Standard Class   
1        IN-2011-47883  01/01/2011  01/08/2011  Standard Class   
2         HU-2011-1220  01/01/2011  01/05/2011    Second Class   
3      IT-2011-3647632  01/01/2011  01/05/2011    Second Class   
4        IN-2011-47883  01/01/2011  01/08/2011  Standard Class   
...                ...         ...         ...             ...   
51285   CA-2014-115427  12/31/2014  01/04/2015  Standard Class   
51286     MO-2014-2560  12/31/2014  01/05/2015  Standard Class   
51287   MX-2014-110527  12/31/2014  01/02/2015    Second Class   
51288   MX-2014-114783  12/31/2014  01/06/2015  Standard Class   
51289   CA-2014-156720  12/31/2014  01/04/2015  Standard Class   

          customer_name      segment             state        country  market  \
0       Toby Braunhardt     Consumer       Constantine        Algeria  Africa   
1           Joseph Holt     Consumer   New South Wales      Australia    APAC   
2         Annie Thurman     Consumer          Budapest        Hungary    EMEA   
3          Eugene Moren  Home Office         Stockholm         Sweden      EU   
4           Joseph Holt     Consumer   New South Wales      Australia    APAC   
...                 ...          ...               ...            ...     ...   
51285        Erica Bern    Corporate        California  United States      US   
51286         Liz Preis     Consumer  Souss-Massa-Draƒ        Morocco  Africa   
51287  Charlotte Melton     Consumer           Managua      Nicaragua   LATAM   
51288     Tamara Dahlen     Consumer         Chihuahua         Mexico   LATAM   
51289     Jill Matthias     Consumer          Colorado  United States      US   

        region  ...         category sub_category  \
0       Africa  ...  Office Supplies      Storage   
1      Oceania  ...  Office Supplies     Supplies   
2         EMEA  ...  Office Supplies      Storage   
3        North  ...  Office Supplies        Paper   
4      Oceania  ...        Furniture  Furnishings   
...        ...  ...              ...          ...   
51285     West  ...  Office Supplies      Binders   
51286   Africa  ...  Office Supplies      Binders   
51287  Central  ...  Office Supplies       Labels   
51288    North  ...  Office Supplies       Labels   
51289     West  ...  Office Supplies    Fasteners   

                                          product_name sales quantity  \
0                                  Tenex Lockers, Blue   408        2   
1                             Acme Trimmer, High Speed   120        3   
2                              Tenex Box, Single Width    66        4   
3                          Enermax Note Cards, Premium    45        3   
4                           Eldon Light Bulb, Duo Pack   114        5   
...                                                ...   ...      ...   
51285  Cardinal Slant-D Ring Binder, Heavy Gauge Vinyl    14        2   
51286          Wilson Jones Hole Reinforcements, Clear     4        1   
51287           Hon Color Coded Labels, 5000 Label Set    26        3   
51288           Hon Legal Exhibit Labels, Alphabetical     7        1   
51289                              Bagged Rubber Bands     3        3   

       discount    profit  shipping_cost  order_priority  year  
0           0.0  106.1400          35.46          Medium  2011  
1           0.1   36.0360           9.72          Medium  2011  
2           0.0   29.6400           8.17            High  2011  
3           0.5  -26.0550           4.82            High  2011  
4           0.1   37.7700           4.70          Medium  2011  
...         ...       ...            ...             ...   ...  
51285       0.2    4.5188           0.89          Medium  2014  
51286       0.0    0.4200           0.49          Medium  2014  
51287       0.0   12.3600           0.35          Medium  2014  
51288       0.0    0.5600           0.20          Medium  2014  
51289       0.2   -

In [43]:
# making the csv file
data.to_csv("final_sales_data.csv",index=False)